src/server/mod.rs

```diff
@@ -946,6 +950,27 @@ fn write_status(socket: WebpushIo) -> MyFuture<()> {
     )
 }
 
+fn write_version_file(socket: WebpushIo) -> MyFuture<()> {
+    let mut file = match File::open("version.json"){
```

Comment:
rust added an fs::read_to_string function that could remove a few lines of code here

Though I've used the include_str! macro in megaphone/pushbox instead. It compiles the version.json contents into the binary. If there's any problem reading the file rustc fails at compile time, so no need for error handling in the app